Simplified/Catalog/NYPLCatalogUngroupedFeed.m

```diff
@@ -176,11 +184,30 @@ - (void)prepareForBookIndex:(NSUInteger)bookIndex
   
   self.greatestPreparationIndex = bookIndex;
   
-  if(self.currentlyFetchingNextURL) return;
+  if(self.books.count - bookIndex > preloadThreshold) {
+    return;
+  }
   
-  if(!self.nextURL) return;
+  [self fetchNextPageWithCompletionHandler:nil];
+}
+
+// This method might trigger a recursive loop, if and only if
+// the fetched feed contains more than one entry AND none of the entries are supported AND nextURL is not nil
+// [NYPLCatalogUngroupedFeed withURL:handler] <-> [NYPLCatalogUngroupedFeed fetchNextPageWithCompletionHandler:]
+// The handler should always be called if it is not nil, in order to exit the recursive loop
+- (void)fetchNextPageWithCompletionHandler:(nullable void (^)(NYPLCatalogUngroupedFeed *category))handler
+{
+  if(self.currentlyFetchingNextURL) {
```

Comment:
I'm not sure we want to call the handler every time we call this method and we are currently fetching... that would result in a disparity of `handler` call compared to the actual fetch.